landoapi/hg.py

```diff
@@ -280,7 +284,7 @@ def clean_repo(self, *, strip_non_public_commits=True):
         except hglib.error.CommandError:
             pass
         try:
-            self.run_hg(["purge", "--all"])
```

Comment:
We wanted the working copy to be clean after calling `clean_repo`, so I assume it was added since we didn't have a need for any VCS ignored files to be retained after cleaning. However we now need to keep at least the objdir and parts of `third_party` in central, so just running `purge` should be good enough for our purposes.